hlapipeline/alignimages.py

```diff
@@ -3,65 +3,173 @@
+    return(totalInputList)
+
+
+def perform_align(input_list):
     """Main calling function.
 
     Parameters
     ----------
     input_list : list
-        List of one or more input image(s) and/or association file(s) to align.
+        List of one or more IPPSSOOTs (rootnames) to align.
 
     Returns
     -------
     Nothing for now.
 
     """
 
+    # Define astrometric catalog list in priority order
+    catalogList = ['GAIADR2', 'GSC241']
+    numCatalogs = len(catalogList)
+
     # 1: Interpret input data and optional parameters
+    print("-------------------- STEP 1: Get data --------------------")
+    imglist = check_and_get_data(input_list)
+    print("\nSUCCESS")
 
     # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
-    imglist = input_list
+    print("-------------------- STEP 2: Filter data --------------------")
+    filteredTable = filter.analyze_data(imglist)
+
+    # Check the table to determine if there is any viable data to be aligned.  The
+    # 'doProcess' column (bool) indicates the image/file should or should not be used
+    # for alignment purposes.
+    if filteredTable['doProcess'].sum() == 0:
+        print("No viable images in filtered table - no processing done.\n")
+        return
+
+    # Get the list of all "good" files to use for the alignment
+    processList = filteredTable['imageName'][np.where(filteredTable['doProcess'])]
+    processList = list(processList) #Convert processList from numpy list to regular python list
+    print("\nSUCCESS")
 
     # 3: Build WCS for full set of input observations
-    refwcs = amutils.build_reference_wcs(imglist)
+    print("-------------------- STEP 3: Build WCS --------------------")
+    refwcs = amutils.build_reference_wcs(processList)
+    print("\nSUCCESS")
 
     # 4: Retrieve list of astrometric sources from database
-    reference_catalog = generate_astrometric_catalog(imglist, catalog='GAIADR2')
-
+    print("-------------------- STEP 4: Detect astrometric sources --------------------")
+    # While loop to accommodate using multiple catalogs
+    doneFitting = False
+    catalogIndex = 0
+    while not doneFitting:
+        reference_catalog = generate_astrometric_catalog(processList, catalog='GAIADR2', existing_wcs=refwcs)
+
+        # The table must have at least MIN_CATALOG_THRESHOLD entries to be useful
+        if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
+            print("Not enough sources found in catalog" + catalogList[catalogIndex])
+            catalogIndex += 1
+            if catalogIndex <= (numCatalogs - 1):
+                print("Try again with the next catalog")
+                continue
+            else:
+                print("Not enough sources found in any catalog - no processing done.")
+                return
+        else:
+            doneFitting = True
+    print("\nSUCCESS")
```

Comment:
There should not be an `else` branch here as the `doneFitting` variable is for when we are really done fitting in Step 7.  The "Try again with the next catalog" `continue` will work once the call noted by Warren is changed to `reference_catalog=generate..., catalog=catalogList[catalogIndex]).`

Reply:
If none of the listed catalogs provide the MIN_CATALOG_THRESHOLD, then we are done.

Reply:
Issue resolved in commit [d8ad5cd](https://github.com/spacetelescope/hlapipeline/pull/36/commits/d8ad5cd11b05df27d4ff7e5580593807a7485309).